src/libvirtApi/domain.js

```diff
@@ -1054,3 +1056,16 @@ export async function domainReplaceSpice({ connectionName, id: objPath }) {
         throw ex; // not-covered: see above
     }
 }
+
+export async function domainAddTPM({ connectionName, vmName }) {
+    const args = ["virt-xml", "-c", `qemu:///${connectionName}`, "--add-device", "--tpm", "default", vmName];
+    return cockpit.spawn(args, { err: "message", superuser: connectionName === "system" ? "try" : null })
+            // RHEL 8 does not support --tpm default; arch (and likely other system setups) fails with
+            // unsupported configuration: TPM version '2.0' is not supported
+            .catch(ex => {
+                if (ex.message?.includes("Unknown TPM backend type") || ex.message?.includes("unsupported configuration"))
+                    console.log("Failed to add TPM:", ex);
+                else
+                    return Promise.reject(ex);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1448-20240219-130934-d81dc524-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)